src/edu/csus/ecs/pc2/exports/ccs/ResultsFile.java

```diff
@@ -263,29 +264,70 @@ else if (record.getNumberSolved() >= median) {
     }
 
     /**
-     * Determine and return award medal color.
+     * Determine whether a team receives a medal or Honor citaion.
      *
      * <pre>
-     * Award is a string with value "gold", "silver", "bronze", "highest honor", "high honor", "honor"
+     * Award is a string with value "gold", "silver", "bronze", "highest honor", "high honor", "honor" if WF type rankings or "ranked"
      * or "honorable" as appropriate.
      * </pre>
      *
-     * @param rankNumber
+     * @param rankNumber,isHighestHonor,isHighHonor,isHonor
      * @param finalizeData
-     * @return
+     * @return medalOrCitation
      */
 
-    private String getAwardMedal(int rankNumber, FinalizeData data, boolean isHighestHonor, boolean isHighHonor, boolean isHonor) {
+    private String getMedalCitation(int rankNumber, FinalizeData data, boolean isHighestHonor, boolean isHighHonor, boolean isHonor) {
 
-        // TODO CCS determine how to assign bronze and ranked
-        if (!(isHighestHonor || isHighHonor || isHonor)) {
-            return HONORABLE;
-        } else if (rankNumber <= data.getGoldRank()) {
+        String medalOrCitation = getAwardMedal(rankNumber, data);
+        if (medalOrCitation == null) {
+            medalOrCitation = getCitation(data, isHighestHonor, isHighHonor, isHonor);
+        }
+        return medalOrCitation;
+    }
+
+    /**
+     * Determine and return award medal color.
+     *
+     * <pre>
+     * Award is a string with value "gold", "silver", "bronze"
+     * or null as appropriate.
+     * </pre>
+     *
+     * @param rankNumber
+     * @param finalizeData
+     * @return medalColor
+     */
+    private String getAwardMedal(int rankNumber, FinalizeData data) {
+
+        if (rankNumber <= data.getGoldRank()) {
             return GOLD;
         } else if (rankNumber <= data.getSilverRank()) {
             return SILVER;
         } else if (rankNumber <= data.getBronzeRank()) {
             return BRONZE;
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Determine what citation team will receive based on new WF rules or are they Ranked if WF rules not used
+     *
+     * <pre>
+     * Award is a string with value "highest honor", "high honor", "honor" if WF type rankings or "ranked"
+     * or "honorable" as appropriate.
+     * </pre>
+     *
+     * @param isHighestHonor,isHighHonor,isHonor
+     * @param finalizeData
+     * @return citation
+     */
+
+     private String getCitation(FinalizeData data, boolean isHighestHonor, boolean isHighHonor, boolean isHonor) {
+
+        // TODO CCS determine how to assign bronze and ranked
```

Comment:
I do not understand what this comment means and why it was added.

Reply:
I can remove, it came in when I copied the previous getAwardMedal() function.

Reply:
Yes, let's get rid of it.  That was when we used to have a value of "B".

Reply:
Updated.